relay-event-schema/src/protocol/span.rs

```diff
@@ -177,6 +165,85 @@ impl Getter for Span {
     }
 }
 
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct SpanData {
+    /// The source code file name that identifies the code unit as uniquely as possible.
+    #[metastructure(field = "code.filepath", pii = "maybe")]
+    pub code_filepath: Annotated<Value>,
+    /// The line number in `code.filepath` best representing the operation.
+    #[metastructure(field = "code.lineno", pii = "maybe")]
+    pub code_lineno: Annotated<Value>,
+    /// The method or function name, or equivalent.
+    ///
+    /// Usually rightmost part of the code unit's name.
+    #[metastructure(field = "code.function", pii = "maybe")]
+    pub code_function: Annotated<Value>,
+    /// The "namespace" within which `code.function` is defined.
+    ///
+    /// Usually the qualified class or module name, such that
+    /// `code.namespace + some separator + code.function`
+    /// form a unique identifier for the code unit.
+    #[metastructure(field = "code.namespace", pii = "maybe")]
+    pub code_namespace: Annotated<Value>,
+
+    /// Other fields in `span.data`.
+    #[metastructure(additional_properties, pii = "true", retain = "true")]
+    other: Object<Value>,
+}
+
+impl SpanData {
+    /// Getter for compatibility with [`Object`].
+    pub fn get(&self, key: &str) -> Option<&Annotated<Value>> {
```

Comment:
Implementing both `Getter` and this `get` seems redundant, could we have a single implementation?

Reply:
We should probably just type out all the `.get()` calls in our code (i.e. replace with named fields), then we can remove this method. I can do that in a follow-up PR.